kmip/services/server/server.py

```diff
@@ -287,17 +287,24 @@ def interrupt_handler(trigger, frame):
         for cipher in auth_suite_ciphers:
             self._logger.debug(cipher)
 
-        self._socket = ssl.wrap_socket(
+        cafile = self.config.settings.get('ca_path')
+        context = ssl.SSLContext(self.ssl_version)
+        context.verify_mode = ssl.CERT_REQUIRED
+        if auth_suite_ciphers:
+            context.set_ciphers(auth_suite_ciphers)
+        if cafile:
+            context.load_verify_locations(cafile)
+        certfile = self.config.settings.get('certificate_path')
+
+        if certfile:
```

Comment:
As with the client, it's worth checking here about how different combinations of not specifying certificate or key behave. For example, the old module-level `wrap_socket()` had a check

```python
    if server_side and not certfile:
        raise ValueError("certfile must be specified for server-side "
                         "operations")
```

so omitting the certificate would have been a hard error.

Reply:
Fixed

Reply:
I think what I had in mind was like

```diff
         if certfile:
             keyfile = self.config.settings.get('key_path')
             context.load_cert_chain(certfile, keyfile=keyfile)
+        else:
+            raise ValueError("certfile must be specified for server-side operations")
```

Reply:
Fixed

Reply:
I instead let the context raise an appropriate error.  This has the benefit of letting us mock the context behavior in the tests.